orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -843,3 +854,306 @@ def _update_onnx_model_initializers(self, state_tensors):
+            partition_info = 'partition_info'
+            states[partition_info] = self._training_session.get_partition_info_map()
+
+        return states
+
+    def _load_model_states(self, training_session_state_dict, state_dict, strict):
+        """Load the model states onto the training session state dictionary"""
+
+        model = 'model'
+
+        # collect all initializer names from the current onnx graph
+        initializer_names = {node.name for node in self._onnx_model.graph.initializer}
+
+        # loaded_initializers dict will be loaded with all the model states from the state dictionary
+        # that are found in the initializer_names dictionary
+        loaded_initializers = {}
+
+        # create an entry for the model in the training session state dictionary
+        if model not in training_session_state_dict:
+            training_session_state_dict[model] = {}
+
+        # copy over model states from the input state dict onto the training session state dict
+        # and copy over model states from the input state dict onto the onnx model
+        for precision, precision_states in state_dict[model].items():
+            if precision not in training_session_state_dict[model]:
+                training_session_state_dict[model][precision] = {}
+
+            for state_key, state_value in precision_states.items():
+                state_value_np = torch.tensor(state_value).numpy()
+                training_session_state_dict[model][precision][state_key] = state_value_np
+                if state_key in initializer_names:
+                    loaded_initializers[state_key] = state_value_np
+                elif strict:
+                    raise RuntimeError("Unexpected key: {} in state_dict[model][{}]".format(state_key, precision))
+
+        # update onnx model from loaded initializers
+        self._update_onnx_model_initializers(loaded_initializers)
+
+    def _load_optimizer_states(self, training_session_state_dict, state_dict):
+        """Load the optimizer states onto the training session state dictionary"""
+
+        optimizer = 'optimizer'
+
+        # create an entry for the optimizer in the training session state dictionary
+        if optimizer not in training_session_state_dict:
+            training_session_state_dict[optimizer] = {}
+
+        # copy over optimizer states from the input state dict onto the training session state dict
+        for model_state_key, optimizer_dict in state_dict[optimizer].items():
+            if model_state_key not in training_session_state_dict[optimizer]:
+                training_session_state_dict[optimizer][model_state_key] = {}
+            for optimizer_state_key, optimizer_state_value in optimizer_dict.items():
+                optimizer_state_value_np = torch.tensor(optimizer_state_value).numpy()
+                training_session_state_dict[optimizer][model_state_key][optimizer_state_key] = optimizer_state_value_np
+
```

Comment:
Why `_load_model_state` calls `self._update_onnx_model_initializers(loaded_initializers)` and `_load_optimizer_states` doesn't?

Reply:
The initializers on the frontend will not have any optimizer states. Since the optimizer states are not part of the onnx_model on the frontend, we don't need to update it using ```self._update_onnx_model_initializers(loaded_initializers)```. Let me know if my understanding is flawed.

Reply:
I think that depends whether `model` passed to `ORTTrainer` is apytorch or onnx model. If user passes models saved with `save_as_onnx`, maybe the optimizer state is there, but I am not 100% sure.

Reply:
Maybe a quick check and issue a warning if optimizer state is found on the `model` as this PR doesnt cover the this scenario, right?

Reply:
Ok sure, I can do this in the subsequent PR.